mbuild/formats/gsdwriter.py

```diff
@@ -49,6 +49,12 @@ def write_gsd(structure, filename, ref_distance=1.0, ref_mass=1.0,
     import gsd.hoomd
 
     xyz = np.array([[atom.xx, atom.xy, atom.xz] for atom in structure.atoms])
+    # Check if we need to shift things to -L/2, L/2
+    box_max = structure.box[:3]/2.
+    wrap = np.greater(xyz, box_max).any()
```

Comment:
Do we want to check to see if any atoms have positions less than -L/2 as well? I guess with the box redefinition from (0, L) to (-L/2, L/2) we really only need to worry about the upper bound, but theoretically I guess particles could be positioned outside of the box in mBuild (not sure if these are wrapped back in when converting to ParmEd or not).

Reply:
I had that check originally to prevent (-L/2, 0) craziness but to quote Rick Cook:
> Programming today is a race between software engineers striving to build bigger and better idiot-proof programs, and the Universe trying to produce bigger and better idiots. So far, the Universe is winning. 

IMHO, if someone is passing in their own box, they *should* know what they are doing. I can add it back in if you still want it :)

Reply:
One of my favorite quotes! I do agree that the user _should_ know what they're doing, but since it is easy for us to check for here, I think we might as well do so. Especially if we factor this out into a `Compound.wrap` method. Then, by the time we get to the XML and GSD writers, the system should already be wrapped, and we can perform a simple check to see if this is the case and raise a warning if it isn't.

Reply:
Sounds good, do you think a simple lower bound/upper bound check is enough? Or should I check per axis in case we get something like `[0, -5, 1, 10, 5, 11]`? I'm leaning towards just an upper lower check since we can figure out the robust way to do a `Compound.wrap` method since that will have many, many more edge cases.

Reply:
I think a simple upper/lower check is probably fine